packages/typescript-fetch-client/templates/api.hbs

```diff
@@ -27,7 +27,7 @@ export const {{className name}}ApiFetchParamCreator = function (configuration?:
 	{{>frag/validateParameter operation=..}}
 	{{/with}}
 			let localVarPath = `{{{../path}}}{{{path}}}`{{#each pathParams}}
-				.replace('{{safe '{'}}{{{serializedName}}}{{safe '}'}}', encodeURIComponent(String({{identifier name}}))){{/each}};
+				.replace('{{safe '{'}}{{{serializedName}}}{{safe '}'}}', encodeURI(String({{identifier name}})).replace('/', '%2F')){{/each}};
```

Comment:
Is this right? We are trying to escape a component of a URI not a whole URI. If there are `?`s in this component they must be escaped otherwise they will be interpreted as starting a query. I think `encodeURIComponent` is right. Didn't this start because you don't like seeing escaped colons? Other languages have different escaping rules for different parts of the URI and maybe that's what we're facing here... maybe a colon doesn't need to be escaped in the path part of the URI. If we can confirm that then we can use the approach of `encodeURIComponent` plus an undoing of colon escaping.

Reply:
School boy error. Also `#` needs to be escaped or it will start the hash...

Of the symbols `encodeURIComponent` escapes that `encodeURI` doesn't, these are the candidates I think we should undo:

```
$ & , : ; @ = +
```

Can you see any issues with unescaping these @karlvr?

Reply:
I've double-checked the RFC to make sure I'm not doing anything dumb like encoding '#', '?' or '/'.

I've added a function to the runtime to do the encoding for us, and elaborated on the reasoning for each included character, and the provenance.